### PR TITLE
Add DSL documentation feature

### PR DESCRIPTION
**Pull request by ofedoren**
@tbrisker, promotion is not my speciality :( Please, let me know your thoughts about how to improve this.

### CODE REVIEW

File: _includes/manuals/nightly/1.2_release_notes.md [hunk 1]
```diff
@@ -4,6 +4,13 @@ This section will be updated prior to the next release.
 
 ### Headline features
 
+#### Templates DSL documentation
+
+Foreman now has documentation for DSL used for templates writing. This should
+make template writing process easier for non-Rubyists as well as for people who
+do know less about certain methods Foreman provides. Documentation is available
```
Comment (tbrisker) [inline _includes/manuals/nightly/1.2_release_notes.md]:
```suggestion
aren't familiar with all the methods Foreman provides. Documentation is available locally
```

File: _includes/manuals/nightly/1.2_release_notes.md [hunk 2]
```diff
@@ -4,6 +4,13 @@ This section will be updated prior to the next release.
...
+under `https://<fqdn>/templates_doc`.
```
Comment (tbrisker) [inline _includes/manuals/nightly/1.2_release_notes.md]: "This documentation will be gradually enhanced in future releases to cover all available methods and macros."

**Review by tbrisker — COMMENTED** [review]
Looks good! added a couple of suggestions inline. Also, was there some discussion on discourse regarding this? if there was it might be a good idea to link to it for further information.

**Comment by ofedoren** [comment]
@tbrisker, thanks a lot! Updated.

**Comment by tbrisker** [comment]
Thanks @ofedoren !

**Pull request merged**